bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/discovery/BuildFileAndVisibilityDrivenProvisioningStrategy.java

```diff
@@ -248,29 +247,17 @@ public Map<BazelProject, Collection<ClasspathEntry>> computeClasspaths(Collectio
 
                 // query for rdeps to find classpath exclusions
                 var projectTargets = activeTargetsPerProject.get(bazelProject);
-                var rdeps = workspace.getCommandExecutor()
-                        .runQueryWithoutLock(
-                            new BazelQueryForLabelsCommand(
-                                    workspaceRoot,
-                                    format(
-                                        "kind(java_library, rdeps(//..., %s))",
-                                        projectTargets.stream().map(BazelLabel::toString).collect(joining(" + "))),
-                                    true,
-                                    format(
-                                        "Querying for reverse dependencies of '%s' for classpath computation",
-                                        bazelProject.getName())))
-                        .stream()
-                        .map(BazelLabel::new)
-                        .collect(toSet());
 
+                var targets = projectTargets.stream().map(BazelLabel::toString).collect(joining(" + "));
                 // get all accessible targets based on visibility
                 Set<BazelLabel> allVisibleTargets = workspace.getCommandExecutor()
                         .runQueryWithoutLock(
                             new BazelQueryForLabelsCommand(
                                     workspaceRoot,
                                     format(
-                                        "kind(java_library, visible(%s, //...))",
-                                        projectTargets.stream().map(BazelLabel::toString).collect(joining(" + "))),
+                                        "kind(java_library, visible(%s, //...)) except kind(java_library, rdeps(//..., %s))",
```

Comment:
Optimization I found, that as far as I can tell, results in the same thing. But could use a second pair of eyes

Reply:
Yeah. good find.